### PR TITLE
Added NEW function for Admin Category

### PR DESCRIPTION
**Pull request by codegracie**
Improved Index.html.erb and added NEW function for Admin Category

### CODE REVIEW

**Comment by kian1213** [comment]
ok

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -1,21 +1,40 @@
 class Admin::CategoriesController < AdminController
-  
   def index
-    @categories = Category.all
+    @categories = Category.order(created_at: :desc)
+    .paginate page: params[:page], per_page: 5
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: indent

File: app/controllers/admin/categories_controller.rb [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
+    end
   end
-  
+
   def edit
   end
-  
+
   def show
     @categories = Category.find params[:id]
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: @category = Category.find params[:id]

File: app/views/admin/categories/new.html.erb [hunk 3]
```diff
@@ -8,6 +8,26 @@
...
             </div>
             <div class="panel-body">
+              <%= form_for [:admin, @category], html: {multipart: true,
+                class: "form-horizontal style-form"} do |f| %>
+                <%= f.label :name %>
+                <%= f.text_field :name, class: "form-control" %>
+                <%= f.label :description %>
+                <%= f.text_area :description, class: "form-control" %>
+                <hr/>
+                <p>Select category photo</p>
```
Comment (MuhammadTamzid) [inline app/views/admin/categories/new.html.erb]: i18n

Comment (codegracie) [inline app/views/admin/categories/new.html.erb]: added

File: app/controllers/admin/categories_controller.rb (hunk 2 above)
Comment (codegracie) [inline app/controllers/admin/categories_controller.rb]: modified

File: app/controllers/admin/categories_controller.rb [hunk 4]
```diff
@@ -1,21 +1,41 @@
...
+  def destroy
+    @categories = Category.find params[:id]
+    @categories.destroy
+    redirect_to admin_categories_path
+  end
+
   def create
+    @category = Category.new category_params
+    if @category.save
+      flash[:success] = "Category saved successfully"
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: i18n

File: app/controllers/admin/categories_controller.rb [hunk 5]
```diff
@@ -1,21 +1,41 @@
...
+      redirect_to admin_categories_path
+    else
+      flash.now[:danger] = "Invalid input"
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: same above

File: app/controllers/admin/categories_controller.rb [hunk 6]
```diff
@@ -1,21 +1,41 @@
...
+      render "new"
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: render :new

File: config/locales/en.yml [hunk 7]
```diff
@@ -152,9 +152,13 @@ en:
...
+        no_items_found: "SEARCH RESULT: THERE ARE NO CATEGORY ITEMS FOUND."
       show: 
         title: Category
       new:
         title: Add New Category
+        new_category: New Category
+        message.create_is_successful: New Category is added successfully!
+        select_photo: Select category photo
       edit:
         title: Edit the Category
```
Comment (MuhammadTamzid) [inline config/locales/en.yml]: add newline

File: app/controllers/admin/categories_controller.rb (hunk 6 above)
Comment (codegracie) [inline app/controllers/admin/categories_controller.rb]: modified

File: app/controllers/admin/categories_controller.rb [hunk 8]
```diff
@@ -1,21 +1,41 @@
...
+    @category = Category.find params[:id]
   end
-  
+
   def show
-    @categories = Category.find params[:id]
+    @category = Category.find params[:id]
+  end
+
+private
```
Comment (dieunb) [inline app/controllers/admin/categories_controller.rb]: indent

File: app/views/admin/categories/new.html.erb [hunk 9]
```diff
@@ -8,6 +8,26 @@
...
+                class: "form-horizontal style-form"} do |f| %>
+                <%= f.label :name %>
+                <%= f.text_field :name, class: "form-control" %>
+                <%= f.label :description %>
+                <%= f.text_area :description, class: "form-control" %>
+                <hr/>
+                <p><%= t ".select_photo" %></p>
+                <div class="form-group">
+                  <div class="col-md-12">
+                    <%= f.file_field :picture, accept: "image/jpeg,image/gif,image/png" %>
```
Comment (dieunb) [inline app/views/admin/categories/new.html.erb]: move to setting whitelist

File: app/controllers/admin/categories_controller.rb (hunk 8 above)
Comment (codegracie) [inline app/controllers/admin/categories_controller.rb]: ok po

File: lib/carrierwave/uploader/extension_whitelist.rb [hunk 10]
```diff
@@ -0,0 +1,3 @@
+def extension_white_list
+  [/jpe?g/, 'gif', 'png']
```
Comment (dieunb) [inline lib/carrierwave/uploader/extension_whitelist.rb]: double quote

**Pull request merged**